### PR TITLE
OGR dataset overhaul

### PR DESCRIPTION
**Pull request by SoerenHoffstedt**
- OGRSourceDatasets: Description parameter added. All parameters can be defined for the whole dataset and for specific layers.
- OGRSource: Parameters for reading the feature collection are constructed from the dataset definition and the layer definition. The layer parameters are prioritized over the dataset parameters.
- OGRSourceDatasets: Dataset listing adjusted to changes. Only returning the layers that are defined in dataset json.
- OGRSourceUtil: Time reading now always checks if the attribute is of type Date/DateTime and if so reads the time from OGR. Before it was only used when the time was filtered with OGR.
- OGRSourceUtil: Additional constructor for moving the params Json. Used by OGRSource that constructs the passed json, avoids copy in OGRSourceUtil.
- OGRSourceUtil: Handles getProvenance for OGRSource and OGRRawSource, therefore gets passed local_identifier as parameter.
- OGRSourceUtil: Added parsing support for MultiSurfaces, but only if they contain polygons.
- Test queries: Adjusted to changes.

### CODE REVIEW

File: src/operators/source/ogr_source.cpp [hunk 1]
```diff
@@ -92,3 +90,90 @@ void OGRSourceOperator::writeSemanticParameters(std::ostringstream &stream) {
...
+        provenanceInfo["uri"] = OGRSourceDatasets::getJsonParameter(layer_provenance, dataset_provenance, "uri");
+
+    constructed_params["provenance"] = provenanceInfo;
+
+    return constructed_params;
+}
+
+Json::Value OGRSourceOperator::getTimeFormatJson(Json::Value &layer_json, Json::Value &dataset_json, const std::string &time_format_name){
+    Json::Value time_format_source;
+    if(layer_json.isMember(time_format_name)) {
```
Comment (michaelmattig) [inline src/operators/source/ogr_source.cpp]: why note use `OGRSourceDatasets::getJsonParameter` here?

Comment (SoerenHoffstedt) [reply src/operators/source/ogr_source.cpp]: Missed that one. Pushed an adjustment.

Comment (SoerenHoffstedt) [reply src/operators/source/ogr_source.cpp]: Just realized too late that the whole function is obsolete. Thought you mean only the if-else. I will adjust it.

File: src/operators/source/ogr_source.cpp [hunk 2]
```diff
@@ -39,24 +52,9 @@ OGRSourceOperator::OGRSourceOperator(int sourcecounts[], GenericOperator *source
...
-    if (provenanceInfo.isObject()) {
-        provenance = Provenance(provenanceInfo.get("citation", "").asString(),
-                                provenanceInfo.get("license", "").asString(),
-                                provenanceInfo.get("uri", "").asString(),
-                                "data.ogr_raw_source." + dataset_name); //todo: does this need the layer_name added?
-    }
-
-    ogrUtil = make_unique<OGRSourceUtil>(loaded_params, provenance);
+    std::string local_id = "data.ogr_source.";
+    local_id.append(dataset_name); //todo: does this need the layer_name added?
```
Comment (michaelmattig) [inline src/operators/source/ogr_source.cpp]: dataset_name is always empty

**Pull request merged**